tests/src/JIT/Regression/JitBlue/GitHub_19550/GitHub_19550.cs

```diff
@@ -0,0 +1,135 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+using System.Threading;
+
+// Test folding of addressing expressions
+
+public class Program
+{
+    struct S
+    {
+        public float f0;
+        public float f1;
+        public float f2;
+        public float f3;
+        public float f4;
+        public float f5;
+        public float f6;
+        public float f7;
+        public float f8;
+        public float f9;
+        public float f10;
+        public float f11;
+        public float f12;
+        public float f13;
+        public float f14;
+        public float f15;
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static unsafe int Test(ref S s, Vector128<float> v, int offset)
+    {
+        int returnVal = 100;
+
+        if (Sse2.IsSupported)
+        {
+            fixed (float* p = &s.f0)
+            {
+                // We need an address aligned on 16 bytes, so we need to add a *float* offset to get there.
+                int alignmentOffset = (0x10 - ((int)p & 0xc)) >> 2;
+                try
+                {
+                    // This is the aligned case.
+                    // We're going to store a scalar at an offset of 2 from the aligned location.
+                    // As it happens, we know that the struct has been initialized to all zeros,
+                    // and the vector passed in was all ones, so now we have a one at offset 2.
+                    Sse2.StoreScalar(p + alignmentOffset + 2, Sse2.Subtract(v, Sse2.LoadAlignedVector128(p + offset + alignmentOffset + 4)));
+
+                    // Now do a non-temporal load from the aligned location.
+                    // That should give us {0, 0, 1, 0}.
+                    Vector128<float> v2 = Sse41.LoadAlignedVector128NonTemporal((byte*)(p + alignmentOffset)).AsSingle();
```

Comment:
We need `Sse41.IsSupported` for this case.

Reply:
Excellent, thanks. I split this out so that we do an `Sse2` load when `Sse41` isn't available.